test/jdk/sun/security/ssl/SSLCipher/SSLSocketNoServerHelloClientShutdown.java

```diff
@@ -96,10 +98,12 @@ private void runTest() throws Exception {
                 serverException = e;
                 log(e.toString());
             } finally {
+                serverLatch.countDown();
```

Comment:
There should be a `latch.await()` at the `getInputStream()` above, as we want to stall the start of the server handshaking until all of the client data has been generated.  (i.e. ClientHello/Alerts X 2).

Reply:
You could also have a different latch here for `join`ing, but that's not as important as the one above.

Reply:
I'm not sure why we need this. We have an infinite timeout now, server will read the packets from the client whenever they become available.

Reply:
I don't think we need this second one here.

Reply:
What would be the first one? Server thread is timing out but I don't think we need  to reverse the logic because of this, we are expecting an exception in the server thread, the client thread should wait until that exception is thrown.

Reply:
See next review.